src/components/Sorting.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="flex w-full justify-between flex-wrap gap-5">
+    <div class="flex gap-3">
+      <button
+        @click="handleClick('title')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByTitle && sortDirection === 'asc',
+          descendant: sortByTitle && sortDirection !== 'asc',
+        }"
+      >
+        Title
+      </button>
+      <button
+        @click="handleClick('text')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByText && sortDirection === 'asc',
+          descendant: sortByText && sortDirection !== 'asc',
+        }"
+      >
+        Description
+      </button>
+      <button
+        @click="handleClick('priority')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByPriority && sortDirection === 'asc',
+          descendant: sortByPriority && sortDirection !== 'asc',
+        }"
+      >
+        Priority
+      </button>
+      <button
+        @click="handleClick('date')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByDate && sortDirection === 'asc',
+          descendant: sortByDate && sortDirection !== 'asc',
+        }"
+      >
+        Date
+      </button>
+    </div>
+
+    <div class="flex gap-2">
+      <button
+        @click="handleSortDirectionChange('asc')"
+        class="ascend-descend-btn bg-emerald-400"
+      >
+        <img src="../assets/Ascending.svg" />
+      </button>
+      <button
+        @click="handleSortDirectionChange('desc')"
+        class="ascend-descend-btn bg-black"
+      >
+        <img src="../assets/Descending.svg" />
+      </button>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import { Todotype } from '../Types/toDo';
+import { SortDirection } from '../Types/SortDirection';
+
+const sortBy = ref('');
+const sortByTitle = ref(false);
+const sortByText = ref(false);
+const sortByPriority = ref(false);
+const sortByDate = ref(false);
+const props = defineProps<{ todos: Todotype[] }>();
+const sortDirection = ref<SortDirection>('asc');
+
+function handleClick(key: string) {
+  sortBy.value = key;
+  updateSortByFlags();
+  sortTodos(sortBy.value, sortDirection.value);
+}
+
+function updateSortByFlags() {
+  sortByTitle.value = sortBy.value === 'title';
+  sortByText.value = sortBy.value === 'text';
+  sortByPriority.value = sortBy.value === 'priority';
+  sortByDate.value = sortBy.value === 'date';
+}
+
+function sortTodos(sortBy: string, sortDirection: SortDirection) {
+  if (sortBy === 'priority') {
+    props.todos.sort((a: Todotype, b: Todotype) => {
+      if (sortDirection === 'desc') {
+        return b.priority - a.priority;
+      } else {
+        return a.priority - b.priority;
+      }
+    });
+    return;
+  }
+
+  props.todos.sort((a: Todotype, b: Todotype) => {
+    if (sortDirection === 'desc') {
+      return b[sortBy].localeCompare(a[sortBy]);
```

Comment:
Please see my comment above related to the `return` and if..else statement, and update this code block based on that.